### PR TITLE
Fix vision

### PR DESCRIPTION
**Pull request by jaagut**
## Necessary checks
- [ ] Update package version
- [ ] Run `catkin build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot
- [ ] Put the PR on our Project board

### CODE REVIEW

File: bitbots_vision/config/visionparams.yaml [hunk 1]
```diff
@@ -26,7 +25,7 @@ bitbots_vision:
     yolo_nms_threshold: 0.4  # Non maximum suppression threshold
     yolo_confidence_threshold: 0.5  # Yolo: confidence threshold
 
-    neural_network_type: 'yolo_ncs2'  # The neural network type that should be used (yolo_opencv, yolo_darknet, yolo_ncs2, yolo_pytorch or dummy)
```
Comment (jaagut) [inline bitbots_vision/config/visionparams.yaml:29]: Revert this...

File: bitbots_vision/bitbots_vision/vision_modules/ros_utils.py [hunk 2]
```diff
@@ -112,8 +113,7 @@ def build_goal_post_array_msg(header, goal_post_msgs):
     # Create goalposts msg
     goal_posts_msg = GoalpostArray()
     # Add header
-    goal_posts_msg.header.frame_id = header.frame_id
-    goal_posts_msg.header.stamp = header.stamp
+    goal_posts_msg.header = header
```
Comment (Flova) [inline bitbots_vision/bitbots_vision/vision_modules/ros_utils.py:116]: The header contains more than the two things. It should be fine, but it is not equivalent.

File: bitbots_vision/bitbots_vision/vision_modules/ros_utils.py [hunk 3]
```diff
@@ -129,10 +129,11 @@ def build_goal_post_msg(goalpost):
     post_msg = Goalpost()
     post_msg.bb = build_bounding_box_2d(goalpost)
     if goalpost.get_rating() is not None:
-        post_msg.confidence = float(goalpost.get_rating())
+        post_msg.confidence.confidence = float(goalpost.get_rating())
+    # TODO: Add attributes
```
Comment (Flova) [inline bitbots_vision/bitbots_vision/vision_modules/ros_utils.py:133]: TODO?

File: bitbots_vision/bitbots_vision/vision_modules/ros_utils.py [hunk 4]
```diff
@@ -158,31 +158,30 @@ def build_ball_msg(ball_candidate):
...
+    :param robots: a list of Robot messages
     :return: RobotArray message
     """
     # Create obstacle msg
-    obstacles_msg = RobotArray()
+    robots_msg = RobotArray()
     # Add header
-    obstacles_msg.header.frame_id = header.frame_id
-    obstacles_msg.header.stamp = header.stamp
+    robots_msg.header = header
```
Comment (Flova) [inline bitbots_vision/bitbots_vision/vision_modules/ros_utils.py:179]: Same here

**Review by Flova — APPROVED** [review]

File: bitbots_vision/bitbots_vision/vision_modules/ros_utils.py (hunk 2 above)
Comment (jaagut) [reply bitbots_vision/bitbots_vision/vision_modules/ros_utils.py:116]: I thought so too, but I checked this yesterday:
```
$ ros2 interface show std_msgs/msg/Header   
# Standard metadata for higher-level stamped data types.
# This is generally used to communicate timestamped data
# in a particular coordinate frame.

# Two-integer timestamp that is expressed as seconds and nanoseconds.
builtin_interfaces/Time stamp
	int32 sec
	uint32 nanosec

# Transform frame with which this data is associated.
string frame_id
```

File: bitbots_vision/bitbots_vision/vision_modules/ros_utils.py (hunk 3 above)
Comment (jaagut) [reply bitbots_vision/bitbots_vision/vision_modules/ros_utils.py:133]: just a reminder...

**Pull request merged**